lib/litmus/type/string.ex

```diff
@@ -140,10 +141,18 @@ defmodule Litmus.Type.String do
     {:ok, params}
   end
 
-  defp length_validate(%__MODULE__{length: length}, field, params)
-       when is_integer(length) and length >= 0 do
-    if Map.has_key?(params, field) && String.length(params[field]) != length do
-      {:error, "#{field} length must be #{length} characters"}
+  defp length_validate(%__MODULE__{length: 0}, field, params) do
+    if params[field] in [nil, ""] do
+      {:ok, params}
+    else
+      {:error, "#{field} length must be 0 characters"}
+    end
+  end
```

Comment:
So this is for the case when a field must have a length of 0, and we consider `nil` to be length 0? I'm assuming this would be a rare case.

Reply:
Lol just read Kyle's comment.